protected/humhub/modules/comment/controllers/CommentController.php

```diff
@@ -78,22 +86,20 @@ public function actionShow()
      */
     public function actionPost()
     {
-        $this->forcePostRequest();
-
-        if (Yii::$app->user->isGuest) {
-            throw new HttpException(403, 'Guests can not comment.');
-        }
-
-        if (!Yii::$app->getModule('comment')->canComment($this->parentContent->content)) {
-            throw new HttpException(403, 'You are not allowed to comment.');
+        if (Yii::$app->user->isGuest || !Yii::$app->getModule('comment')->canComment($this->parentContent->content)) {
+            throw new ForbiddenHttpException(
+                Yii::t(
+                    'CommentModule.controllers_CommentController',
+                    'You are not allowed to comment.'
+                )
+            );
         }
 
         $message = Yii::$app->request->post('message');
         $files = Yii::$app->request->post('fileList');
 
         if (empty(trim($message)) && empty($files)) {
-            // do not create empty comments
-            return '';
+            throw new BadRequestHttpException('The comment is wrong');
```

Comment:
Instead of using `The comment is wrong` you should think of using `Comment could not be created`

Reply:
... and users will be created topics "why I can't create comment ?" ;)

P.S. this action used by javascript HubHub code. Before PR javascript get 200OK if comment empty. Now 400 error with flash message.

 TODO: Javascript code should not be passed empty comments.